### DISCUSSION TITLE
Converting a PDF email loses the background and it then takes on the color of whatever application you use to view.

### DISCUSSION THREAD

**Discussion by pbracken** [dlemstra/Magick.NET#1669] [Help] [answered]
I'm trying to convert a few emails as pdfs to image files but they appear to lose the background and then the background is taking on background of the view app which in certain cases makes it unreadable. I've tried setting BackgroundColor and FillColor and other colors but nothing makes a difference. I'm sure there is some setting I'm missing so any help would be appreciated. I have attached a sample email PDF and the subsequent image 

```
using ImageMagick;

foreach(var filePath in Directory.EnumerateFiles("C:\\Users\\PatrickBracken\\Downloads\\Proof of Ownership\\Proof of Ownership"))
{
	if (filePath.EndsWith(".pdf"))
	{
		var settings = new MagickReadSettings()
		{
			//Doesn't appear to do anything in this case
			BackgroundColor = MagickColors.White,
			//Doesn't appear to do anything in this case
			FillColor = MagickColors.White,
		};
		var pdfPages = new MagickImageCollection(filePath, settings);
		var horizontal = pdfPages.AppendVertically();
		horizontal.Write(filePath.Replace(".pdf", ".png"));
	}
	
```

![0 - Testing background](https://github.com/dlemstra/Magick.NET/assets/10732038/b25745c9-c4aa-4bf1-87a1-ebba47aebebd)
[0 - Testing background.pdf](https://github.com/user-attachments/files/16110331/0.-.Testing.background.pdf)

**Comment by dlemstra** [comment]
You can turn the alpha channel off by going through the `pdfPages` and turning the alpha channel off with `pdfPage.HasAlpha = false;`.

**Reply by pbracken** [reply]
Ok that does fix that issue, now my test looks good but the other ones have black backgrounds in certain sections. Any ideas for this? I've tried setting different colors to White but that doesn't appear to have any affect. Maybe I am missing one.

**Comment by dlemstra** [comment, marked as answer]
Another option would be this:

```C#
image.BackgroundColor = MagickColors.White;
image.Alpha(AlphaOption.Remove);
```

**Reply by pbracken** [reply]
That was it! Thank you so much for the help.
